sequencer/src/chain_config.rs

```diff
@@ -1,58 +1,73 @@
-use crate::state::FeeAmount;
+use crate::{
+    options::parse_size,
+    state::{FeeAccount, FeeAmount},
+};
+use clap::Args;
 use committable::{Commitment, Committable};
 use derive_more::{From, Into};
-use ethers::types::U256;
+use ethers::types::{Address, U256};
 use itertools::Either;
 use sequencer_utils::impl_to_fixed_bytes;
 use serde::{Deserialize, Serialize};
+use std::str::FromStr;
 
 #[derive(Default, Hash, Copy, Clone, Debug, Deserialize, Serialize, PartialEq, Eq, From, Into)]
 pub struct ChainId(U256);
 
 impl_to_fixed_bytes!(ChainId, U256);
 
-impl From<u16> for ChainId {
-    fn from(id: u16) -> Self {
+impl From<u64> for ChainId {
+    fn from(id: u64) -> Self {
         Self(id.into())
     }
 }
 
+impl FromStr for ChainId {
+    type Err = <u64 as FromStr>::Err;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        Ok(u64::from_str(s)?.into())
+    }
+}
+
 /// Global variables for an Espresso blockchain.
-#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, Serialize, Deserialize)]
+#[derive(Args, Debug, Clone, Copy, PartialEq, Eq, Hash, Serialize, Deserialize)]
 pub struct ChainConfig {
     /// Espresso chain ID
-    chain_id: ChainId,
+    #[clap(long, env = "ESPRESSO_SEQUENCER_CHAIN_ID", default_value = "0")]
+    pub chain_id: ChainId,
     /// Maximum size in bytes of a block
-    max_block_size: u64,
+    #[clap(long, env = "ESPRESSO_SEQUENCER_MAX_BLOCK_SIZE", default_value = "30mb", value_parser = parse_size)]
+    pub max_block_size: u64,
     /// Minimum fee in WEI per byte of payload
-    base_fee: FeeAmount,
+    #[clap(long, env = "ESPRESSO_SEQUENCER_BASE_FEE", default_value = "0")]
+    pub base_fee: FeeAmount,
+    /// Fee contract address on L1.
+    ///
+    /// This is optional so that fees can easily be toggled on/off, with no need to deploy a
+    /// contract when they are off. In a future release, after fees are switched on and thoroughly
+    /// tested, this may be made mandatory.
+    #[clap(long, env = "ESPRESSO_SEQUENCER_FEE_CONTRACT_PROXY_ADDRESS")]
+    pub fee_contract: Option<Address>,
+    /// Account that receives sequencing fees.
+    #[clap(
+        long,
+        env = "ESPRESSO_SEQUENCER_FEE_BURN_ACCOUNT",
+        default_value = "0x0000000000000000000000000000000000000000"
+    )]
+    pub fee_burn_account: FeeAccount,
```

Comment:
Do we need a configurable burn account or could we just use burn to the address if the fee_contract is None? If this is too implicit then suggest we at least make it impossible to have both a burn address and the fee contract address set.

Reply:
Oh, the intent was that this parameter is orthogonal to the fee contract being set: when fees are paid in Espresso, we always burn to this address, and then, whenever we deploy a fee contract, it is responsible for figuring out how to handle "withdrawals" from the burn address...ie what to do retroactively with burned tokens.

So it's important that this is set if the fee contract is not so we don't lose tokens forever before we've deployed the contract, but it is also meaningful if the contract is set. I can add comments to clarify

Reply:
See [2ac26c0](https://github.com/EspressoSystems/espresso-sequencer/pull/1390/commits/2ac26c031bb6c55f818660fe482c8ecbc8232093)

Reply:
Thanks for the explanations. I think I was really confused by the term "burn" and hadn't read the implementation in detail yet. In Ethereum burn usually means making the funds inaccessible by sending it to an address for which no-one has the private keys (or somewhat more involved actually making them vanish). Can we call it fee recipient account?

Reply:
Done [4d34e25](https://github.com/EspressoSystems/espresso-sequencer/pull/1390/commits/4d34e25bd278d5cd707c3a18f444c84829ecaa51)